Application/EdFi.Ods.Api/Caching/ExpiringConcurrentDictionaryCacheProvider.cs

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using log4net;
+
+namespace EdFi.Ods.Api.Caching
+{
+    public class ExpiringConcurrentDictionaryCacheProvider : ICacheProvider
+    {
+        private readonly IDictionary<string, object> _cacheDictionary = new ConcurrentDictionary<string, object>();
+
+        private readonly ILog _logger = LogManager.GetLogger(typeof(ExpiringConcurrentDictionaryCacheProvider));
+
+        private Timer _timer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExpiringConcurrentDictionaryCacheProvider" /> class using the
+        /// specified recurring expiration period.
+        /// </summary>
+        /// <param name="expirationPeriod">The recurring expiration period for all of the entries in the cache.</param>
+        public ExpiringConcurrentDictionaryCacheProvider(TimeSpan expirationPeriod)
+        {
+            _timer = new Timer(CacheExpired, null, expirationPeriod, expirationPeriod);
```

Comment:
This class is explicitly instantiated as a dependency of a singleton, so you should be safe with the timer consistently remaining static.  I really dislike the use of a timer here and this seems to be yet another unintended side effect of rolling our own caching.  Given this is caching related and any of the usual issues with timers should also result in the cache being dropped this is probably ok as a short term solution until we determine a more consistent way to handle the caching.